src/AwesomizeExpress.re

```diff
@@ -0,0 +1,40 @@
+/* Private Functions */
+let badRequest = Express.Response.StatusCode.BadRequest;
+
+let ok = Express.Response.StatusCode.Ok;
+
+let beltMapToJson = map =>
+  Belt.Map.String.toArray(map) |> Js.Dict.fromArray |> Js.Json.object_;
+
+let errorMapToJson = map =>
+  Belt.Map.String.map(map, x =>
+    switch (x) {
+    | None => Js.Json.null
+    | Some(str) => Js.Json.string(str)
+    }
+  )
+  |> beltMapToJson;
+
+let sendJson = (res, statusCode, json) =>
+  res
+  |> Express.Response.status(statusCode)
+  |> Express.Response.sendJson(json)
+  |> Js.Promise.resolve;
+
+/* Public Functions */
+/* @TODO - Awesomizer needs to support a context and this integration needs
+   to accept a function which can generate that context from the request
+   object. */
+let make = (~schema, ~decoder, ~encoder, ~handler, _, req, res) =>
+  Express.Request.asJsonObject(req)
+  |> Awesomize.make(schema)
+  |> Awesomize.Result.fold(
+       err => sendJson(res, badRequest, errorMapToJson(err)),
```

Comment:
This is a matter of style, and I'm not sure what we should go with.

```reason
err => err |> errMapToJson |> sendJson(res, badRequest)
```
```reason
Json.Promise.then_(json => json |> encoder |> sendJson(res, ok)
```
or...
```reason
err => errorMapToJson(err) |> sendJson(res, badRequest)
```
```reason
Json.Promise.then_(json => encoder(json) |> sendJson(res, ok)
```
or ...
```reason
err => sendJson(res, badRequest, errorMapToJson(err)),
```
```reason
Json.Promise.then_(json => sendJson(res, ok, encoder(json)))
```
Or, we could use [rationale]
```reason
open Rationale.Function.infix;
Json.Promise.then_ ||> encoder ||> sendJson(res, ok)
```

 @acbarnett @carmonw  @travistrue2008 @dvisztempacct
[rationale]:https://github.com/jonlaing/rationale

Reply:
I don't really have a preference, but I'd rather not have a dependency on something that pertains to style; not only will this add unnecessary overhead but it will add points of inconsistency if we don't enforce the usage of rationale.